java/core/src/test/java/sleeper/core/record/process/status/ProcessRunsTest.java

```diff
@@ -190,4 +190,15 @@ public void shouldIgnoreExtraFinishedStatus() {
         assertThat(runs.isFinished()).isTrue();
     }
 
+    @Test
+    void shouldNotCreateProcessRunIfStatusUpdateNotFlaggedAsStartOfRun() {
+        // Given
+        ProcessStatusUpdate notStartedUpdate = () -> Instant.parse("2022-09-24T09:23:30.001Z");
+
+        // When
+        ProcessRuns runs = runsFromUpdates(notStartedUpdate);
+
+        // Then
+        assertThat(runs.getRunList()).isEmpty();
+    }
 }
```

Comment:
I can see that these changes will make the behaviour we need easier, but we can test the behaviour we want here.

We can test the behaviour when we implement ProcessRunStartedUpdate twice, where one of the implementations lets you flag it as not actually starting the run.

We can also test that we can retrieve a status update from the run that isn't a started or finished update. Right now I don't think it will actually be added to the run. It would be useful to retrieve the update by a class, with a generic method that returns an instance of that class.